src/Serialization.h

```diff
@@ -5,6 +5,16 @@
 
 namespace Halide {
 
+/**
+ * Serialize a Halide pipeline into the given data buffer.
+ * params is an optional map which can be used to bind external parameters to objects in the pipeline by name
+ */
+void serialize_pipeline(const Pipeline &pipeline, std::vector<uint8_t> &data, std::map<std::string, Internal::Parameter> &params);
```

Comment:
Food for future thought: this is in the `Halide` namespace and so is a public API... but it uses `Internal::Parameter`, so it really isn't quite public. Would be nice to figure out how to avoid needing to mix these.

Reply:
Good point ... it seems like we may need to revise this a little bit so its only using public class interfaces.

Reply:
Updated